pgjdbc/src/test/java/org/postgresql/test/jdbc2/BlobTest.java

```diff
@@ -200,26 +203,85 @@ void uploadBlob_NATIVE() throws Exception {
   void markResetStream() throws Exception {
     assertTrue(uploadFile(TEST_FILE, NATIVE_STREAM) > 0);
 
+    long oid;
     try (Statement stmt = con.createStatement()) {
       try (ResultSet rs = stmt.executeQuery("SELECT lo FROM testblob where id = '/test-file.xml'")) {
         assertTrue(rs.next());
+        oid = rs.getLong(1);
+      }
+    }
+
+    LargeObjectManager lom = ((PGConnection) con).getLargeObjectAPI();
+    try (LargeObject blob = lom.open(oid)) {
+      InputStream bis = blob.getInputStream();
+      assertEquals('<', bis.read());
+      bis.mark(4);
+      assertEquals('?', bis.read());
+      assertEquals('x', bis.read());
+      assertEquals('m', bis.read());
+      assertEquals('l', bis.read());
+      bis.reset();
+      assertEquals('?', bis.read());
+    }
 
-        LargeObjectManager lom = ((PGConnection) con).getLargeObjectAPI();
+    try (LargeObject blob = lom.open(oid)) {
+      blob.seek(4);
+      InputStream bis = blob.getInputStream();
+      assertEquals('l', bis.read());
+      bis.reset();
+      assertEquals('l', bis.read());
+      assertEquals(' ', bis.read());
+      bis.mark(4);
+      assertEquals('v', bis.read());
+      assertEquals('e', bis.read());
+      bis.reset();
+      assertEquals('v', bis.read());
+    }
+  }
 
-        long oid = rs.getLong(1);
-        LargeObject blob = lom.open(oid);
-        InputStream bis = blob.getInputStream();
-
-        assertEquals('<', bis.read());
-        bis.mark(4);
-        assertEquals('?', bis.read());
-        assertEquals('x', bis.read());
-        assertEquals('m', bis.read());
-        assertEquals('l', bis.read());
-        bis.reset();
-        assertEquals('?', bis.read());
+  @Test
+  void skip() throws Exception {
+    long expectedBigSkip;
+    if (TestUtil.haveMinimumServerVersion(con, ServerVersion.v9_3)) {
+      expectedBigSkip = 4398046442492L;
+    } else {
+      expectedBigSkip = 2147417083;
+    }
+
+    // We need a blob large enough to use more than one buffer.
+    LargeObjectManager lom = ((PGConnection) con).getLargeObjectAPI();
+    long loid = lom.createLO();
+    try (LargeObject blob = lom.open(loid, LargeObjectManager.WRITE)) {
+      byte[] buf = new byte[1024];
+      for (byte i = 0; i < 96; i++) {
+        Arrays.fill(buf, i);
```

Comment:
I would prefer random data since 11111 22222 333 might mask bugs in case read accesses a wrong buffer item

Reply:
The test has to know which value is expected. Random data would be difficult to verify by hand.

As there are only 256 possible byte values, there will always be duplicates in the data, so you cannot prove it wasn't accidentally reading a different byte that happened to have the same value.